Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -242,6 +242,12 @@ public override void Reset()
         /// <inheritdoc />
         public override void Disable()
         {
+            // Input System adds a gaze provider component on the main camera, which needs to be removed when the input system is disabled/removed.
+            // Otherwise the component would keep references to dead objects.
+            // Unity's way to remove component is to destroy it. Immediate is important, otherwise order of deinitialization will be arbitrary.
+            MixedRealityInputSystemProfile profile = ConfigurationProfile as MixedRealityInputSystemProfile;
+            UnityEngine.Object.DestroyImmediate(CameraCache.Main.gameObject.GetComponent(profile.PointerProfile.GazeProviderType.Type));
```

Comment:
[From the Unity docs](https://docs.unity3d.com/ScriptReference/Object.DestroyImmediate.html):

>This function should only be used when writing editor code since the delayed destruction will never be invoked in edit mode.

Also:

>You are strongly recommended to use Destroy instead.

Does the order of deinitialization cause issues that can't be resolved in other ways?